src/plugins/validation/oas3/semantic-validators/security-definitions-ibm.js

```diff
@@ -0,0 +1,191 @@
+        path,
+        authId: key
+      });
+    } else if (auths.indexOf(type) === -1) {
+      errors.push({
+        message:
+          '`type` must have one of the following types: `apiKey`, `oauth2`, `http`, `openIdConnect`',
+        path,
+        authId: key
+      });
+    } else {
+      //apiKey validation
+      if (type === API_KEY) {
+        const authIn = security.in;
+        if (!authIn) {
+          errors.push({
+            message:
+              "apiKey authorization must have required 'in' property, valid values are 'query' or 'header' or 'cookie'.",
+            path,
+            authId: key
+          });
+        } else if (
+          authIn !== 'query' &&
+          authIn !== 'header' &&
+          authIn != 'cookie'
+        ) {
+          errors.push({
+            message:
+              "apiKey authorization must have required 'in' property, valid values are 'query' or 'header' or 'cookie'.",
+            path,
+            authId: key
+          });
+        }
+      }
+
+      if (type === API_KEY && !security.name) {
+        errors.push({
+          message:
+            "apiKey authorization must have required 'name' string property. The name of the header or query parameter to be used.",
+          path,
+          authId: key
+        });
+      }
+      // oauth2 validation
+      else if (type === OAUTH2) {
+        const flows = security.flows;
+        const tokenUrl = security.tokenUrl;
+
+        if (!flows) {
+          errors.push({
+            message:
+              "oauth2 authorization must have required 'flows'. Valid values are 'implicit', 'password', 'clientCredentials' or 'authorizationCode'", /////recieved
+            path,
+            authId: key
+          });
+        } else if (
+          !flows.implicit &&
+          !flows.authorizationCode &&
+          !flows.authorizationCode &&
+          !flows.clientCredentials
+        ) {
+          errors.push({
+            message:
+              "oauth2 authorization must have required 'flows'. Valid values are 'implicit', 'password', 'clientCredentials' or 'authorizationCode'", /////recieved
+            path,
+            authId: key
+          });
+        } else if (flows && flows.implicit) {
+          const authorizationUrl = flows.implicit.authorizationUrl;
+          if (!authorizationUrl) {
+            errors.push({
+              message:
+                "oauth2 authorization implicit flow must have required 'authorizationUrl' parameter if type is `implicit` or `authorizationCode`.", /////////exepected
+              path,
+              authId: key
+            });
+          } else if (
+            type === OAUTH2 &&
+            flows &&
+            flows.implicit &&
+            !flows.implicit.scopes
+          ) {
+            errors.push({
+              message:
+                "'scopes' is required property type object. The available scopes for the OAuth2 security scheme.",
```

Comment:
This message is confusing.  I think it should be reworded along the lines of the other messages, e.g.:
> oauth2 authorization implicit flow must have required 'scopes' property.